lib/src/async_environment.dart

```diff
@@ -122,6 +126,7 @@ class AsyncEnvironment {
   AsyncEnvironment({bool sourceMap = false})
       : _modules = {},
         _globalModules = null,
+        _allModules = [],
```

Comment:
Why the mix of representation between `_globalModules = null` and `_allModules = []`? Is this just an optimistic optimization in hopes that global modules will be rare?

Reply:
Exactly.